test/test_circular_buffer.jl

```diff
@@ -5,7 +5,8 @@ using Base.Test
 cb = CircularBuffer{Int}(5)
 @test length(cb) == 0
 @test capacity(cb) == 5
-@test_throws ArgumentError first(cb)
+# throws ArgumentError on v0.4 and BoundsError on v0.5
+@test_throws Exception first(cb)
```

Comment:
I meant in the test, not a comment - you should check for as specific an exception type as you can